src/MiniCover/Reports/Clover/CloverReport.cs

```diff
@@ -0,0 +1,198 @@
+
+            var document = new XDocument(
+                new XDeclaration("1.0", "utf-8", null),
+                CreateCoverageElement(result, hits)
+            );
+
+            var xmlWriterSettings = new XmlWriterSettings
+            {
+                Indent = true
+            };
+
+            var path = Path.GetDirectoryName(output);
+
+            if(!string.IsNullOrEmpty(path))
+            {
+                Directory.CreateDirectory(Path.GetDirectoryName(output));
+            }
+
+            using (StreamWriter sw = File.CreateText(output))
+            using (XmlWriter writer = XmlWriter.Create(sw, xmlWriterSettings))
+            {
+                document.WriteTo(writer);
+            }
+        }
+
+        private static XElement CreateCoverageElement(InstrumentationResult result, IDictionary<int, int> hits)
+        {
+            var timestamp = DateTimeOffset.UtcNow.ToUnixTimeSeconds();
+
+            return new XElement(
+                XName.Get("coverage"),
+                new XAttribute(XName.Get("generated"), timestamp),
+                new XAttribute(XName.Get("clover"), "4.1.0"),
+                CreateProjectElement(result, timestamp, hits)
+            );
+        }
+
+        private static XElement CreateProjectElement(InstrumentationResult result, long timestamp, IDictionary<int, int> hits)
+        {
+            return new XElement(
+                XName.Get("project"),
+                new XAttribute(XName.Get("timestamp"), timestamp),
+                new XAttribute(XName.Get("name"), result.SourcePath),
+                CreateMetricsElement(CountProjectMetrics(result, hits)),
+                CreatePackagesElement(result, hits)
+            );
+        }
+
+        private static IEnumerable<XElement> CreatePackagesElement(InstrumentationResult result, IDictionary<int, int> hits)
+        {
+            return result.Assemblies.Select(assembly => new XElement(
+                XName.Get("package"),
+                new XAttribute(XName.Get("name"), assembly.Name),
+                CreateMetricsElement(CountPackageMetrics(assembly, hits)),
+                CreateFilesElement(assembly, hits)
+            ));
+        }
+
+        private static IEnumerable<XElement> CreateFilesElement(InstrumentedAssembly assembly, IDictionary<int, int> hits)
+        {
+            return assembly.SourceFiles.Select(file => new XElement(
+                XName.Get("file"),
+                new XAttribute(XName.Get("name"), Path.GetFileName(file.Key)),
+                new XAttribute(XName.Get("path"), file.Key),
+                CreateMetricsElement(CountFileMetrics(file.Value, hits)),
+                CreateClassesElement(file.Value.Instructions, hits),
+                CreateLinesElement(file.Value.Instructions, hits)
+            ));
+        }
+
+        private static IEnumerable<XElement> CreateClassesElement(ICollection<InstrumentedInstruction> instructions, IDictionary<int, int> hits)
```

Comment:
it should be IEnumerable<> 
https://msdn.microsoft.com/en-us/library/system.collections.ienumerable.aspx
If you are using the only enumeration on the list it is behind IEnumerable<>. ICollection is responsibility for manipulation on the list.

Reply:
ok